src/models/link.es6.js

```diff
@@ -5,6 +5,34 @@ class Link extends Base {
   constructor(props) {
     delete props.selftext_html;
 
+    if (props.promoted) {
```

Comment:
Talked in person.  Basically the reason for not using a separate model is because this is how it works internally to reddit and if it was a different model then we'd have to handle initialization on every page `Link/Ad` are used. (comments page, listing, etc).  Doing it this removes that edge case.